spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeltaWrite.java

```diff
@@ -567,43 +562,16 @@ public void close() throws IOException {
       }
     }
 
-    private PartitioningWriter<InternalRow, DataWriteResult> newInsertWriter(
-        Table table,
-        SparkFileWriterFactory writerFactory,
-        OutputFileFactory fileFactory,
-        Context context) {
-      long targetFileSize = context.targetDataFileSize();
-
-      if (table.spec().isPartitioned() && context.fanoutWriterEnabled()) {
-        return new FanoutDataWriter<>(writerFactory, fileFactory, table.io(), targetFileSize);
-      } else {
-        return new ClusteredDataWriter<>(writerFactory, fileFactory, table.io(), targetFileSize);
-      }
-    }
-
-    private PartitioningWriter<InternalRow, DataWriteResult> newUpdateWriter(
-        Table table,
-        SparkFileWriterFactory writerFactory,
-        OutputFileFactory fileFactory,
-        Context context) {
-      long targetFileSize = context.targetDataFileSize();
-
-      if (table.spec().isPartitioned()) {
-        // use a fanout writer for partitioned tables to write updates as they may be out of order
-        return new FanoutDataWriter<>(writerFactory, fileFactory, table.io(), targetFileSize);
-      } else {
-        return new ClusteredDataWriter<>(writerFactory, fileFactory, table.io(), targetFileSize);
-      }
+    private PartitioningWriter<InternalRow, DataWriteResult> newDataWriter(
```

Comment:
Can you help me understand why this no longer supports the explicit fanout writer option?

I think it's because we are always adding a shuffle after the merge or update anyway so it never matters?

Reply:
We always add a local sort as it is required to write out position deletes. That's why we always add a sort by partition columns if the table is partitioned.